### PR TITLE
libs/libc/obstack: implement ptr and int growing functions

### PR DESCRIPTION
**Pull request by Cynerd**
## Summary
This API is defined by GNU and implemented in other LibCs, such as Musl.

## Testing

Custom samv7 based board.

### CODE REVIEW

**Review by lupyuen — APPROVED** [review]

File: include/obstack.h [hunk 1]
```diff
@@ -135,11 +132,50 @@
...
+ *   There is no check if there is enough room and thus it is easy to cause
+ *   buffer overrun. Use only when you are sure that there is enough room!
+ *
+ * Input Parameters:
+ *   h: pointer to the handle used to grow the object.
+ *   ptr: pointer to be added
+ *
+ ****************************************************************************/
+#define obstack_ptr_grow_fast(h, ptr) \
+  ({ \
```
Comment (xiaoxiang781216) [inline include/obstack.h:152]: change to `do while`, since `({})` is a gcc special extension.

File: include/obstack.h [hunk 2]
```diff
@@ -135,11 +132,50 @@
...
+ *   buffer overrun. Use only when you are sure that there is enough room!
+ *
+ * Input Parameters:
+ *   h: pointer to the handle used to grow the object.
+ *   data: integer to be added
  *
  ****************************************************************************/
-#define obstack_1grow_fast(h, data) (*((h)->next_free++) = (data))
+#define obstack_int_grow_fast(h, data) \
+  ({ \
```
Comment (xiaoxiang781216) [inline include/obstack.h:173]: ditto

File: include/obstack.h [hunk 3]
```diff
@@ -117,11 +117,8 @@
  *   h: pointer to the handle used to grow the object.
  *   size: number of bytes
  *
- * Returned Value:
- *   The new address just after the end of the currently growing object.
- *
  ****************************************************************************/
-#define obstack_blank_fast(h, size) ((h)->next_free += (size))
+#define obstack_blank_fast(h, size) ((void)((h)->next_free += (size)))
```
Comment (xiaoxiang781216) [inline include/obstack.h:121]: let' prepend blank line between the comment and ALL macro definition.

**Review by xiaoxiang781216 — CHANGES_REQUESTED** [review]

File: include/obstack.h (hunk 1 above)
Comment (Cynerd) [reply include/obstack.h:152]: Ok, but there are already palces where `({})` are used such as `nuttx/list.h` and `nuttx/spinlock.h` (although those can be disabled). I have nothing against changing that only that not using that extension might be hard either way.

File: include/obstack.h (hunk 3 above)
Comment (Cynerd) [reply include/obstack.h:125]: Ok

File: include/obstack.h (hunk 2 above)
Comment (Cynerd) [reply include/obstack.h]: ...

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/obstack.h (hunk 1 above)
Comment (xiaoxiang781216) [reply include/obstack.h]: it's better to avoid the compiler special extension as much as we can.

**Pull request merged**